e2e-test/cbas_azure_e2e_test.py

```diff
@@ -0,0 +1,240 @@
+    headers = {
+        "Authorization": f"Bearer {azure_token}",
+        "accept": "application/json"
+    }
+
+    response = requests.get(uri, headers=headers)
+    status_code = response.status_code
+
+    if status_code != 200:
+        logging.error(response.text)
+        exit(1)
+
+    response = json.loads(response.text)
+    if response['run_sets'][0]['state'] != 'COMPLETE':
+        logging.error(f"Submission '{run_set_id}' not in 'COMPLETE' state. Current state: {response['run_sets'][0]['state']}.")
+        exit(1)
+
+    logging.info(f"Submission '{run_set_id}' status: COMPLETE")
+
+
+# ---------------------- Start Workflows Azure E2E test ----------------------
+logging.info("Starting Workflows Azure E2E test...")
+
+# Create workspace
+logging.info("Creating workspace...")
+workspace_id, workspace_name = create_workspace(billing_project_name, azure_token, rawls_url)
+
+# Create WORKFLOWS_APP and CROMWELL_RUNNER apps in workspace
+create_app(workspace_id, leo_url, 'WORKFLOWS_APP', 'WORKSPACE_SHARED', azure_token)
+create_app(workspace_id, leo_url, 'CROMWELL_RUNNER_APP', 'USER_PRIVATE', azure_token)
+
+# sleep for 5 minutes to allow workspace to provision and apps to start up
+logging.info("Sleeping for 5 minutes to allow workspace to provision and apps to start up...")
+time.sleep(5 * 60)
+
+# Upload data to workspace
+# check that WDS is ready; if not exit the test after 10 minutes of polling
+logging.info(f"Checking to see if WDS app is ready to upload data for workspace {workspace_id}...")
+wds_url = poll_for_app_url(workspace_id, 'WDS', 'wds', azure_token, leo_url)
+if wds_url == "":
+    logging.error(f"WDS app not ready or errored out for workspace {workspace_id}")
+    exit(1)
+upload_wds_data_using_api(wds_url, workspace_id, "e2e-test/resources/cbas/cbas-e2e-test-data.tsv", "test-data")
+
+# check that CBAS is ready; if not exit the test after 10 minutes of polling
+logging.info(f"Checking to see if WORKFLOWS app is ready in workspace {workspace_id}...")
+cbas_url = poll_for_app_url(workspace_id, 'WORKFLOWS_APP', 'cbas', azure_token, leo_url)
+if cbas_url == "":
+    logging.error(f"WORKFLOWS app not ready or errored out for workspace {workspace_id}")
+    exit(1)
+
+# check that Cromwell Runner is ready; if not exit the test after 10 minutes of polling
+logging.info(f"Checking to see if CROMWELL_RUNNER app is ready in workspace {workspace_id}...")
+cromwell_url = poll_for_app_url(workspace_id, 'CROMWELL_RUNNER_APP', 'cromwell-runner', azure_token, leo_url)
+if cromwell_url == "":
+    logging.error(f"CROMWELL_RUNNER app not ready or errored out for workspace {workspace_id}")
+    exit(1)
+
+# create a new method
+method_id = create_cbas_method(cbas_url, workspace_id)
+method_version_id = get_method_version_id(cbas_url, method_id)
+
+# submit workflow to CBAS
+run_set_id = submit_no_tasks_workflow(cbas_url, method_version_id)
+
+# sleep for 2 minutes to allow submission to finish
```

Comment:
TOL: As far as flakiness, what if we polled WDS for up to X minutes rather than 1 hit or miss at the 2 minute mark (then it also _could_ finish faster)? This would also give some more timing flexibility in the future when cromwell runner is auto-launched by workflows app (unless we still want to manually launch cromwell runner in the e2e test).